apps/dotcom-worker/src/lib/utils/roomOpenMode.ts

```diff
@@ -0,0 +1,17 @@
+import { ROOM_OPEN_MODE, RoomOpenMode } from '@tldraw/dotcom-shared'
+import { exhaustiveSwitchError, lns } from '@tldraw/utils'
+import { Environment } from '../types'
+
+export async function getSlug(env: Environment, slug: string | null, roomOpenMode: RoomOpenMode) {
+	if (!slug) return null
+	switch (roomOpenMode) {
+		case ROOM_OPEN_MODE.READ_WRITE:
+			return slug
+		case ROOM_OPEN_MODE.READ_ONLY:
+			return await env.READONLY_SLUG_TO_SLUG.get(slug)
+		case ROOM_OPEN_MODE.READ_ONLY_LEGACY:
+			return lns(slug)
```

Comment:
what are the drawbacks of doing something along the lines of `( || lns(slug)` and keeping the URL as `/v/`?

Off the top of my head, the main issue is maybe that that would mean lns URLs would work for new rooms (although they wouldn't be shared anywhere), but we could prevent that by doing something like:
```ts
const storedSlug = await env.READONLY_SLUG_TO_SLUG.get(slug)
if (storedSlug) return storedSlug

const legacySlug = lns(slug)
const isLegacyRoom = !(await env.SLUG_TO_READONLY_SLUG.get(legacySlug))
if (isLegacyRoom) return legacySlug
return null
```

Reply:
Now that we locked down room creation the only thing it gives us is a bit more traceability if something goes wrong. We could maybe reuse `/v` and loose quite a bit of complexity. WDYT @mimecuvalo?

Reply:
Yeah, I think a couple things:
- it opens up the path of eventually deprecating the older path, if we, say after a year (prbly longer), see that there's no traffic there. 
- As Mitja mentions, there's more traceability to narrowing down an issue with older paths vs. newer ones. If they start getting conflated it'll start being harder to debug/maintain.
- it felt icky in the sense of a threat of a collision being present. It's not possible actually to have a collision between these two, it's just that it felt like, eng-wise, if were migrating to, say, similar-but-different hashing functions we would have to account for this and either create a versioning scheme or different path. It's only a quirk that the lns logic is silly enough where it isn't possible. So I suppose I'm saying it feels more correct to separate these two.

I hear the argument for keeping parity, but the prefix of the url's aren't 'pretty' or memorable anyway ¯\\_(ツ)_/¯ 
Only `r` is arguably memorable, but the rest are perhaps incidental so it didn't feel too important to keep it the same.

Reply:
I feel like some of this could be mitigated with a migration to write the old lns IDs into the KV store.

if we do want to have two separate URLs, could we make the new one still be a single-character?